src/components/Carousel/Carousel.tsx

```diff
@@ -61,28 +63,56 @@ export const Carousel = ({
     }
   }, [autoPlay, embla])
 
+  useEffect(() => {
+    if (!embla) return
+    // Re-initialize the carousel when slides change
+    embla.reInit()
+  }, [embla, childrens])
+
+  useEffect(() => {
+    // We need to set the ref only when the component is mounted
+    // This prevents incorrect calculations of the width of the slides
+    setIsVisible(true)
+  }, [])
+
   const renderSlides = useMemo(() => {
     return childrens.map((child, i) => (
-      <Box className='embla__slide' key={i} flex='0 0 100%' marginLeft={4}>
+      <Box
+        className='embla__slide'
+        position='relative'
+        minWidth={0}
+        key={i}
+        flex='0 0 100%'
+        paddingLeft='10px'
+      >
         {child}
       </Box>
     ))
   }, [childrens])
 
   return (
-    <Box ref={viewportRef} className='embla' overflow='hidden'>
-      <Box className='embla__container' display='flex'>
-        {renderSlides}
+    <Box className='embla'>
+      <Box className='embla__viewport' ref={isVisible ? viewportRef : null} overflow='hidden'>
+        <Box
+          className='embla__container'
+          display='flex'
+          alignItems='flex-start'
+          transition='height 0.2s'
+          height='auto'
+          marginLeft='calc(10px * -1)'
+        >
+          {renderSlides}
+        </Box>
       </Box>
-      {(showDots || showArrows) && (
-        <Flex justifyContent='space-between' alignItems='center' mt={4}>
+      {(showDots || showArrows) && scrollSnaps.length > 1 && (
```

Comment:
If you want to save 4 chars
```suggestion
      {(showDots || showArrows) && scrollSnaps.length && (
```

Reply:
Disregard me am derp